### ISSUE TITLE
[next-auth][error][NO_SECRET] with provided secret in config

### ISSUE THREAD

**Issue by gyurobenjamin** [nextauthjs/next-auth#5565]
### Environment

  System:
    OS: macOS 12.3
    CPU: (10) arm64 Apple M1 Pro
    Memory: 1.11 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v16.16.0/bin/yarn
    npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
  Browsers:
    Brave Browser: 103.1.41.96
    Chrome: 106.0.5249.103
    Firefox: 103.0.1
    Safari: 15.4
  npmPackages:
    next: ^12.2.0 => 12.3.1 
    next-auth: ^4.10.3 => 4.10.3 
    react: ^17.0.2 => 17.0.2 

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

Getting server error with provided secret:
Browser:
```
Server error
There is a problem with the server configuration.

Check the server logs for more information.
```
Logs:
```
[next-auth][error][NO_SECRET] 
https://next-auth.js.org/errors#no_secret
```

NextAuth config:

```
export default NextAuth({
  session: {
    strategy: 'jwt',
  },
  providers: [
    CredentialsProvider({
      type: 'credentials',
      credentials: {},
      authorize: async () => {
        return {
        };
      },
    }),
  ],
  pages: {
    signIn: '/auth/signin',
  },
  secret: '5/IFQsBlMyvdiZ+BSIOZb8YEWN73FtvbXRfDep10gyk=',
})
```

### How to reproduce

See description

### Expected behavior

Not getting this error with provided secret.

**Comment by balazsorban44** [comment]
The env variable `NEXTAUTH_SECRET` is prioritized (and recommended) over the `secret` option: https://github.com/nextauthjs/next-auth-example/blob/e140b26f2cf14309a0bbf6ffdeeb7cf4d5469ff5/.env.local.example#L2

Likely it is set to an empty string and as such the assertion will fail here: https://github.com/nextauthjs/next-auth/blob/fe7aaeded872a5ca17c01fa835e42cedf2bc7e83/packages/next-auth/src/core/lib/assert.ts#L54

**Comment by gyurobenjamin** [comment]
Yes, I had to set this var in the pipeline to perform cypress tests. Docs suggested it was supposed to work defining the `secret` in the config. It was not set to an empty string.
